pkg/deployment/deployment.go

```diff
@@ -313,3 +256,111 @@ func ConditionalDeploy(
 	return err
 
 }
+
+// addServiceExtraMounts adds the service configs as ExtraMounts to aeeSpec
+func addServiceExtraMounts(
+	ctx context.Context,
+	helper *helper.Helper,
+	aeeSpec dataplanev1.AnsibleEESpec,
+	service dataplanev1.OpenStackDataPlaneService,
+) (dataplanev1.AnsibleEESpec, error) {
+
+	client := helper.GetClient()
+	baseMountPath := path.Join(ConfigPaths, service.Name)
+
+	for _, cmName := range service.Spec.ConfigMaps {
```

Comment:
I guess it's possible that both of these parameters are empty, since they are optional. Maybe we can exit early in that case before starting all of this logic?

```
if len(service.Spec.ConfigMaps) == 0 && len(service.Spec.Secrets) == 0 {
    return aeeSpec, nil
}
```

Because this function is handling both cases, we can't check one without the other otherwise we might skip required logic. But maybe an early out if both are empty will save a little time.

It's not a huge deal though, if you're ready to merge it I wouldn't worry about it now. I think a for loop over a nil value would be fairly quick anyway. We're probably only talking a difference of a couple of ms.